metrics-exporter-prometheus/src/exporter/builder.rs

```diff
@@ -125,6 +125,7 @@ impl PrometheusBuilder {
         interval: Duration,
         username: Option<String>,
         password: Option<String>,
+        http_method: Option<hyper::Method>,
```

Comment:
I'd prefer seeing this be `use_post: bool`. I'm not a huge fan of adding another parameter that everyone has to care about even if they just want the default... but I'd prefer seeing this be `use_post: bool`.

Otherwise, this is unconstrained and means that the user can screw up and pass a method that isn't valid (`GET`, etc) and that they actually need to import the `hyper::Method` type in the first place.